File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
   let left = 0, right = height.length - 1;
  let maxWater = 0;

  while (left < right) {
  
    const width = right - left;
    const minHeight = Math.min(height[left], height[right]);
    maxWater = Math.max(maxWater, width * minHeight);

    
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return maxWater;




    
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int maxarea=0;

        while(l<r){ 
           int h = Math.min(height[l], height[r]);
            int w = r - l;

            maxarea=Math.max(maxarea,h*w);

            if(height[l]<height[r]){
                l++;
            }
             else{
                r--;
             }


        }
        return maxarea;
    }
}